app/src/main/java/ch/sdp/vibester/database/DataGetter.kt

```diff
@@ -49,45 +49,58 @@ class DataGetter @Inject constructor() {
 
     /**
      * This function increments a specific int field of a user in the database
-     * @param userID the id of the user which is being updated
-     * @param newVal (Int) the increment value of the field that is being updated
-     * @param fieldName the field name of the field that is being updated
+     * @param userID
+     * @param newVal integer value to update
+     * @param fieldName field name to updated
      */
     fun updateRelativeFieldInt(userID: String, newVal: Int, fieldName: String) {
-        if(!TestMode.isTest()) {
-            dbUserRef.child(userID).child(fieldName)
-                .get().addOnSuccessListener { t ->
-                    if(!TestMode.isTest()) {
-                        updateFieldInt(userID, (t.value as Long?)!!.toInt() + newVal, fieldName)
-                    }
-                }
-        }
+        dbUserRef.child(userID).child(fieldName)
+            .get().addOnSuccessListener { t ->
+                var sum  = newVal
+                if(t.value != null){sum += (t.value as Long?)!!.toInt()}
+                updateFieldInt(userID, sum, fieldName)
+            }
+
     }
 
     /**
      * This function sets the best between current and given in argument value
      * @param userID the id of the user which is being updated
-     * @param newVal (Int) the new value of the field that is being compared and updated
-     * @param fieldName the field name of the field that is being updated
+     * @param newVal integer value to update
+     * @param fieldName field name to updated
      */
     fun updateBestFieldInt(userID: String, newVal: Int, fieldName: String) {
-        if(!TestMode.isTest()) {
-            dbUserRef.child(userID).child(fieldName)
-                .get().addOnSuccessListener { t ->
-                    if(!TestMode.isTest()) {
-                        updateFieldInt(userID, maxOf((t.value as Long?)!!.toInt(), newVal), fieldName)
-                    }
+        dbUserRef.child(userID).child(fieldName)
+            .get().addOnSuccessListener { t ->
+                var maxValue = newVal
+                if(t.value != null){maxValue = maxOf((t.value as Long?)!!.toInt(), newVal)}
+                updateFieldInt(userID, maxValue, fieldName)
                 }
-        }
+    }
+
+    /**
+     * Update the best value of subfield
+     * @param userID
+     * @param newVal integer value to update
+     * @param fieldName field name to updated
+     * @param subFieldName subfield name to updated
+     */
+    fun updateBestSubFieldInt(userID: String, newVal: Int, fieldName: String, subFieldName: String) {
```

Comment:
The three update functions are very similar, refactoring them can not only fix code climate issues but also boost the test coverage. Do you think is doable?

Reply:
I have refactored the database calls. They have become cleaner, the coverage remained same, and did not fall!